src/Java.Interop.Tools.JavaTypeSystem/Adapters/JavaXmlApiImporter.cs

```diff
@@ -0,0 +1,370 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Java.Interop.Tools.JavaTypeSystem.Models;
+
+namespace Java.Interop.Tools.JavaTypeSystem
+{
+	public class JavaXmlApiImporter
+	{
+		public static JavaTypeCollection ParseString (string xml, JavaTypeCollection? collection = null)
+		{
+			var doc = XDocument.Parse (xml, LoadOptions.SetBaseUri | LoadOptions.SetLineInfo);
+
+			return Parse (doc, collection);
+		}
+
+		public static JavaTypeCollection Parse (TextReader reader, JavaTypeCollection? collection = null)
+		{
+			var doc = XDocument.Load (reader, LoadOptions.SetBaseUri | LoadOptions.SetLineInfo);
+
+			return Parse (doc, collection);
+		}
+
+		public static JavaTypeCollection Parse (string filename, JavaTypeCollection? collection = null)
+		{
+			var doc = XDocument.Load (filename, LoadOptions.SetBaseUri | LoadOptions.SetLineInfo);
+
+			return Parse (doc, collection);
+		}
+
+		static JavaTypeCollection Parse (XDocument doc, JavaTypeCollection? collection = null)
+		{
+			collection ??= new JavaTypeCollection ();
+
+			var root = doc.Root;
+
+			if (root is null)
+				throw new Exception ("Invalid XML file");
+
+			collection.ApiSource = root.XGetAttributeOrNull ("api-source");
+			collection.Platform = root.XGetAttributeOrNull ("platform");
+
+			var packages = new List<JavaPackage> ();
+
+			foreach (var elem in root.Elements ()) {
+				switch (elem.Name.LocalName) {
+					case "package":
+						packages.Add (ParsePackage (elem, collection));
+						break;
+				}
```

Comment:
Should we emit a warning or something when `elem.Name.LocalName != "package"`?

The only other element I know of which could *plausibly* show up here is `<enum/>`, e.g. in `src/Mono.Android/obj/Release/monoandroid10/android-30/mcw/api.xml.fixed`:

```xml
  <enum name="Android.Hardware.Fingerprints.FingerprintAuthenticationFlags" />  
</api>
```

Anything else is likely to be a mistake, e.g. if there's an `/api/class` element.

Reply:
I don't know that adding warnings for things like this is useful.  On one hand, they should never happen because we're consuming our own output from `class-parse`.  On the other hand, if it did happen no one would notice because we generate so many other warnings in the binding process.